src/rust-crypto/backup.ts

```diff
@@ -106,6 +106,27 @@ export class RustBackupManager extends TypedEventEmitter<RustBackupCryptoEvents,
         };
     }
 
+    /**
+     * Checks if the provided backup info matches the given private key.
+     *
+     * @param info - The backup info to check.
+     * @param backupDecryptionKey - The `BackupDecryptionKey` private key to check against.
+     * @returns `true` if the private key can decrypt the backup, `false` otherwise.
+     */
+    private backupInfoMatchesBackupDecryptionKey(
```

Comment:
can we use this in `isKeyBackupTrusted`?  I think we just need

```
const backupMatchesSavedPrivateKey = backupKeys?.decryptionKey !== undefined && backupInfoMatchesBackupDecruptionKey(info, backupKeys.decryptionKey);
```

Reply:
not anymore because as per previous comment we move it to a standalone function

Reply:
It's not a big deal, but I don't understand why that makes any difference.

Here's what I'm suggesting, in full:

```typescript
    /**
     * Determine if a key backup can be trusted.
     *
     * @param info - key backup info dict from {@link MatrixClient#getKeyBackupVersion}.
     */
    public async isKeyBackupTrusted(info: KeyBackupInfo): Promise<BackupTrustInfo> {
        const signatureVerification: SignatureVerification = await this.olmMachine.verifyBackup(info);
        const backupKeys: RustSdkCryptoJs.BackupKeys = await this.olmMachine.getBackupKeys();
        const decryptionKey = backupKeys?.decryptionKey;
        const backupMatchesSavedPrivateKey = !!decryptionKey && backupInfoMatchesBackupDecryptionKey(info, decryptionKey);
        return {
            matchesDecryptionKey: backupMatchesSavedPrivateKey,
            trusted: signatureVerification.trusted(),
        };
    }
```

does that not work?

Reply:
Didn't know about the `!!` trick.
Yes it should work, looks better, updated.